### PR TITLE
cr overrides - replicas

### PR DESCRIPTION
**Pull request by tchughesiv**
addresses some of https://github.com/font/gatekeeper-operator/issues/19

Signed-off-by: tchughesiv <tchughesiv@gmail.com>

### CODE REVIEW

**Comment by tchughesiv** [comment]
@font i had to remove these annotations in order to deploy to an openshift 4.5 cluster. would you like me to add them back?
```
      annotations:	
        container.seccomp.security.alpha.kubernetes.io/manager: runtime/default
```

**Review by JAORMX — COMMENTED** [review]
@tchughesiv have you tried using a privileged SCC for the pods? that would allow you to set security attributes such as the seccomp profile.

**Comment by tchughesiv** [comment]
@JAORMX i did not... even if it worked, i'm not sure that would be a wise deployment requirement though.

**Comment by JAORMX** [comment]
Using the privileged SCC doesn't make the pod be privileged. And it would allow you to set:
1. a fine-grained seccomp profile (or the default which is better than nothing)
2. reduced capabilities

If you remove the usage of the seccomp setting, the pod will run without confinement.

**Comment by tchughesiv** [comment]
@JAORMX ill revert the annotations for this PR since it's a separate issue... and we can open an issue to discuss level of privilege in openshift separately.

**Comment by font** [comment]
> @font i had to remove these annotations in order to deploy to an openshift 4.5 cluster. would you like me to add them back?
> 
> ```
>       annotations:	
>         container.seccomp.security.alpha.kubernetes.io/manager: runtime/default
> ```

@tchughesiv Yes please, as that's being handled by #26.

**Comment by tchughesiv** [comment]
@font done

File: controllers/gatekeeper_controller_test.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+*/
+
+package controllers
+
+import (
+	"testing"
+
+	operatorv1alpha1 "github.com/font/gatekeeper-operator/api/v1alpha1"
+	"github.com/openshift/library-go/pkg/manifest"
+	"github.com/stretchr/testify/assert"
```
Comment (font) [inline controllers/gatekeeper_controller_test.go:24]: Any reason not to use the gomega package assertions https://onsi.github.io/gomega/#making-assertions that are already set up by the operator-sdk?

Comment (tchughesiv) [reply controllers/gatekeeper_controller_test.go:24]: @font no reason... don't know how i missed that. will make the switch now

Comment (tchughesiv) [reply controllers/gatekeeper_controller_test.go]: @font its been switched to gomega

File: controllers/gatekeeper_controller_test.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test",
+			Namespace: "testns",
+		},
+	}
+	// test default audit replicas
+	auditManifest, err := getManifest(auditFile)
+	g.Expect(err).ToNot(HaveOccurred())
+	g.Expect(auditManifest).ToNot(BeNil())
+	testManifestReplicas(t, auditManifest, int64(1))
```
Comment (font) [inline controllers/gatekeeper_controller_test.go:43]: I don't expect the default replica value (1 in this case) to change in upstream gatekeeper, but we should consider adding a constant var for these.

**Review by font — APPROVED** [review]
Thanks @tchughesiv ! This LGTM. Please rebase now that #29 merged.

File: controllers/gatekeeper_controller_test.go (hunk 2 above)
Comment (tchughesiv) [reply controllers/gatekeeper_controller_test.go]: @font done

**Pull request merged**